docs/schema.md

```diff
@@ -921,6 +921,68 @@ If this section is not specified, users see a warning when deploying the app to
 an Istio-enabled environment. The [`ISTIO_ENABLED`](#type-istio_enabled) type
 indicates whether Istio is enabled on the cluster.
 
+### assistedClusterCreation
+
+Use `assistedClusterCreation` to specify constraints for cluster creation as part of app deployment. These constraints determine whether a new cluster creation is allowed as part of deployment; and if yes, what are the node constraints.
```

Comment:
"; and if yes, what are the node constraints." -->
", and allow declaration of a specific node configuration for creation."

The intro is probably a good place to call out that `assistedClusterCreation` is optional if cluster creation is enabled. Partners can use it to specify a specific node configuration, but otherwise one will be generated according to defaults or their specified `clusterConstraints`, if declared. The node config defined here will override the one suggested by `clusterConstraints`.